src/safeds/data/tabular/transformation/_k_nearest_neighbors_imputer.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from numpy import nan
+
+from safeds._utils import _structural_hash
+from safeds._validation import _check_bounds, _check_columns_exist, _ClosedBound
+from safeds.data.tabular.containers import Table
+from safeds.exceptions import TransformerNotFittedError
+
+from ._table_transformer import TableTransformer
+
+if TYPE_CHECKING:
+    from sklearn.impute import KNNImputer as sk_KNNImputer
+
+
+class KNearestNeighborsImputer(TableTransformer):
+    """
+    The KNearestNeighborsImputer replaces missing values in given Columns with the mean value of the K-nearest neighbors.
+
+    Parameters
+    ----------
+    neighbor_count:
+        The number of neighbors to consider when imputing missing values.
+    column_names:
+        The list of columns used to impute missing values. If 'None', all columns are used.
+    value_to_replace:
+        The placeholder for the missing values. All occurrences of`missing_values` will be imputed.
+    """
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Dunder methods
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def __init__(
+        self,
+        neighbor_count: int,
+        *,
+        column_names: str | list[str] | None = None,
+        value_to_replace: float | str | None = nan,
```

Comment:
```suggestion
        value_to_replace: float | str | None = None,
```

The user should not see that `nan` is used internally.